bold_test.py

```diff
@@ -12,3 +12,43 @@
 def test_render(content, html):
     bold = BoldBlock(content)
     assert html == bold.render()
+
+
+@pytest.mark.parametrize(
+    "content, start, end, sub_content",
+    [
+        # Bold and Italic (Either head or tail has 3 asterisks and total asterisks > 6)
+        ('***test***', 0, 10, '*test*'),
+        ('****test***', 0, 11, '**test*'),
+        ('***test****', 0, 11, '*test**'),
+        ('*****test***', 0, 12, '***test*'),
+        ('***test*****', 0, 12, '*test***'),
+        # Bold only (Otherwise from above, head and tail has at least 2 asterisks)
+        ('**test**', 0, 8, 'test'),
+        ('***test**', 0, 9, '*test'),
+        ('**test***', 0, 9, 'test*')
+    ]
+)
+def test_parse(content, start, end, sub_content):
+    (start_index, end_index, bold_block) = BoldBlock.parse(content)
+    assert start_index == start
+    assert end_index == end
+    assert bold_block._content == sub_content
+
+
+@pytest.mark.parametrize(
+    "content",
+    [
+        # not match
+        ('test'),
```

Comment:
add cases like: 
"\**abc**"
"**abc\***"
"**abc\**"

Reply:
> 
> 
> add cases like:
> "*_abc_*"
> "**abc***"
> "**abc**"

[*](2)abc[*](3) -> will become abc* as you see in this github markdown, it's covered in the above test function,

[*](2)abc[*](2) ->is the regular bold, it's also covered in the above test function.